src/utils/config.hpp

```diff
@@ -100,17 +102,16 @@ class Config {
     int embrakes;
   } sensors;
 
-//  private:
+  struct Factory {
+#define CREATOR_FUNCTION_POINTERS(module, interface) \
```

Comment:
Seems like now we are using the word "module" in two different meanings now?

1. Sensors / Navigation / State machine / ...
2. The namespace under which an interface class is declared

Any ideas for renaming something to avoid the conflict?

Reply:
All of those are supposed to mean more or less the same:
1. directories in `src/` (sw modules with corresponding threads/mains)
2. namespaces
3. data structures in `src/data/data.hpp`
4. data structures in `src/utils/config.hpp`
5. slack channels / sw teams / clickup projects (approximately)

I can see the name conflicts but I do not mind them personally. However, if we decide to resolve the conflict here, should we also resolve it between all of those above (especially between points 1-4)?

Reply:
I wouldn't be resolving all of them unless you have an idea on how to resolve it. It doesn't seem too problematic yet.
However, `module` on this line doesn't help to understand that one is supposed to pass in a namespace (exactly because the word "module" is used so loosely elsewhere. A better name might be `interface_namespace` but that's a bit long. You could also add a short comment explaining what `module` is here. But I guess it's not too bad so you can leave it for now if you can't think of anything.